### ISSUE TITLE
Custom dataset support issue

### ISSUE THREAD

**Issue by foundnom** [HengWeiBin/CenterFusionDetect3D#3]
The original centerfusion puts all radar_pc in train.json, though this project puts radar_pc logic in generic_dataset, it makes harder to support a custom dataset because one must have generic_dataset changed.
Would you consider add an option to have an original centerfusion-style radar_pc method?

**Comment by HengWeiBin** [comment]
You're right, placing the `radar_pc` logic in `generic_dataset` makes custom dataset integration less flexible. In early October, I updated the code to handle radar data loading within the custom dataset class itself (see [nuScenes.py](https://github.com/HengWeiBin/CenterFusionDetect3D/blob/d775cbeb89a4565417f7038311a082d9af1b43e0/src/lib/dataset/datasets/nuscenes.py#L131)). This allows for dataset-specific preprocessing, especially for radar data which can vary significantly between vendors and modules. Is this the specific issue you're trying to address?

**Comment by foundnom** [comment]
> You're right, placing the `radar_pc` logic in `generic_dataset` makes custom dataset integration less flexible. In early October, I updated the code to handle radar data loading within the custom dataset class itself (see [nuScenes.py](https://github.com/HengWeiBin/CenterFusionDetect3D/blob/d775cbeb89a4565417f7038311a082d9af1b43e0/src/lib/dataset/datasets/nuscenes.py#L131)). This allows for dataset-specific preprocessing, especially for radar data which can vary significantly between vendors and modules. Is this the specific issue you're trying to address?

Yeah, and by updating to the newest version it's much better to support custom datasets. Thanks a lot that really solve my prob!
